Python/bytecodes.c

```diff
@@ -5027,6 +5027,35 @@ dummy_func(
             assert(tstate->tracing || eval_breaker == FT_ATOMIC_LOAD_UINTPTR_ACQUIRE(_PyFrame_GetCode(frame)->_co_instrumentation_version));
         }
 
+        inst(CHECK_ITERABLE, (iterable_st -- iterable_st)) {
+            PyObject *iterable = PyStackRef_AsPyObjectBorrow(iterable_st);
+            PyTypeObject *type = Py_TYPE(iterable);
+            if (oparg == 0) {
+                // Sync case, similar to GET_ITER
+                if (type->tp_iter == NULL) {
+                    if (!PySequence_Check(iterable)) {
```

Comment:
Consider simplifying this:
```suggestion
                if (type->tp_iter == NULL && !PySequence_Check(iterable)) {
```

Though, what's the point of the extra check for a sequence here? Do we need one for the async case as well?

Reply:
This extra check is copied from ``PyObject_GetIter``.
In case of sequence object, ``GET_ITER`` will make an iterator by ``PySeqIter_New`` function.
It seems to be useful that ``CHECK_ITERABLE`` will have the same logic.

In the async case there is only one option, we call ``type->tp_as_async->am_aiter`` function.

See ``GET_ITER`` and ``GET_AITER`` bytecode implementations.